lib/diaspora/shareable.rb

```diff
@@ -57,7 +65,7 @@ def owned_or_visible_by_user(user)
         with_visibility.where(
           visible_by_user(user).or(arel_table[:public].eq(true)
                                      .or(arel_table[:author_id].eq(user.person_id)))
-        ).select("DISTINCT #{table_name}.*")
+        ).distinct_wrapper
```

Comment:
So, this is only for tests? I don't like changing the code (and maybe making it less readable) only for tests.

Also, if you change your test to
```ruby
expect(StatusMessage.where_person_is_mentioned(@bob).ids).to match_array([post1.id, post2.id])
```
it works with both databases without changing anything on the code. Also this would actually test if you see the correct posts, and not two other (maybe wrong) posts.